site/content/languages/_index.md

```diff
@@ -18,14 +18,62 @@ Ex. If your source is in Go, you might compile it with TinyGo.
 Below are notes wazero contributed so far, in alphabetical order by language.
 
 * [TinyGo](tinygo) Ex. `tinygo build -o X.wasm -target=wasi X.go`
+* [Rust](rust) Ex. `rustc -o X.wasm --target wasm32-wasi X.rs`
 
-wazero is a runtime that embeds in Golang applications, not a web browser. As
+wazero is a runtime that embeds in Go applications, not a web browser. As
 such, these notes bias towards backend use of WebAssembly, not browser use.
 
 Disclaimer: These are not official documentation, nor represent the teams who
 maintain language compilers. If you see any errors, please help [maintain][1]
 these and [star our GitHub repository][2] if they are helpful. Together, we can
 make WebAssembly easier on the next person.
 
+## Concurrency
+
+WebAssembly's virtual machine has no concept of processor or core. Similarly,
+it lacks instructions for atomics, wait/notify or barriers commonly used to
+implement parallel processing.
+
+For example, a compiler targeting [WASI][3], generates a `_start` function
+corresponding to `main` in the original source code. When the WebAssembly
+runtime calls `_start`, it remains on the same thread of execution until that
+function completes.
+
+Concretely, if using wazero, a Wasm function call remains on the calling
+goroutine until it completes.
+
+In summary, while true that host functions can do anything, including launch
+processes, Wasm binaries compliant with [WebAssembly Core 2.0][4] cannot do
+anything in parallel, unless they use non-standard instructions or conventions
+not yet defined by the specification.
+
```

Comment:
Suggest mentioning Asyncify here.

Reply:
oh this is the top-level page, so I did mention it in the tinygo one (so many things in common between languages, there's some top-level content now). That said, I think I need to also mention asyncify in the internals part of tinygo, too.

Reply:
actually you are right, it isn't just used by tinygo. I'll put it here!